src/Voron/Data/Lookups/Lookup.cs

```diff
@@ -613,11 +620,15 @@ private void FreePageFor(ref CursorState stateToKeep, ref CursorState stateToDel
         {
             // let's reduce the height of the tree entirely...
             DecrementPageNumbers(ref parent);
-
+            
```

Comment:
I haven't found a test case for this but I believe we've to call OnKeyRemoval to remove the key that will removed from the tree (but a term will persist on disk without any reference to it...), right?